core/extension/gdextension_interface.h

```diff
@@ -283,6 +285,12 @@ typedef struct {
 	GDExtensionClassCreateInstance create_instance_func; // (Default) constructor; mandatory. If the class is not instantiable, consider making it virtual or abstract.
 	GDExtensionClassFreeInstance free_instance_func; // Destructor; mandatory.
 	GDExtensionClassGetVirtual get_virtual_func; // Queries a virtual function by name and returns a callback to invoke the requested virtual function.
+	// Can be used instead of `get_virtual_func`. Returns user data (non-null) that will be passed back when calling `call_virtual_func`.
+	// Returning `null` from this function signals to Godot that the virtual function is not overridden.
+	// Data provided to this function must live until the GDExtensionInstance is freed, but should be deleted when `free_instance_func` is called.
```

Comment:
Adjusted the language a bit. Let me know what you think.

Reply:
Hm. Thinking about it some more, I'm not sure I still like my suggestion either :-)

The GDExtension really doesn't know when the `GDExtensionInterface` will be freed. I think all it can do is keep the data alive until the GDExtension is deinitialized. I'll put that in a new suggestion in a moment...